### PR TITLE
[BE1][BE2] Add timestamp to rumors

### PR DESCRIPTION
**Pull request by DanielTavaresA**
To be able to add partial ordering to messages, we want to add a vector clock to each rumor. For this, jsonRPC of Rumor has been changed with the addition of a RumorState object, that will represent the rumor state in which a rumor has been sent

### CODE REVIEW

File: docs/protocol.md [hunk 1]
```diff
@@ -982,6 +985,12 @@ RPC
     "params": {
         "sender_id": "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=",
         "rumor_id": 1,
+        "timestamp" : {
+            "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=": 3,
```
Comment (sgueissa) [inline docs/protocol.md:989]: can you change   "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=": 3, to   "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=": 1, same for the other place

File: docs/protocol.md [hunk 2]
```diff
@@ -1127,6 +1141,12 @@ Response in case of success
     {
       "sender_id": "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=",
       "rumor_id": 1,
+      "timestamp" : {
+        "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=": 3,
```
Comment (sgueissa) [inline docs/protocol.md:1145]: same

File: docs/protocol.md [hunk 3]
```diff
@@ -1142,6 +1162,12 @@ Response in case of success
     {
       "sender_id": "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=",
       "rumor_id": 2,
+      "timestamp" : {
+        "J9fBzJV70Jk5c-i3277Uq4CmeL4t53WDfUghaK0HpeM=": 3,
```
Comment (sgueissa) [inline docs/protocol.md:1166]: same but change to 2

**Review by sgueissa — APPROVED** [review]
lgtm

**Review by K1li4nL — APPROVED** [review]

**Pull request merged**